nvOC_miner_update.sh

```diff
@@ -385,16 +785,90 @@ echo
 echo
 sleep 2
 
+function compile-TPccminer {
+  echo "Compiling tpruvot ccminer"
+  echo " This could take a while ..."
+  if [[ $CUDA_VER == "8.0" ]]
+  then
+    if [[ -d ${NVOC_MINERS}/TPccminer/src/ ]]
+    then
+      cd ${NVOC_MINERS}/TPccminer/src/ 
+      if [[ $(git config --get remote.origin.url) != $TPccminer_repo ]]
+      then
+        echo "Wrong repo, clean up"
+        rm -rf ${NVOC_MINERS}/TPccminer/src
+        echo "Clone tpruvot repo from $TPccminer_repo"
+        git clone $TPccminer_repo ${NVOC_MINERS}/TPccminer/src/
+      fi
+    else
+      echo "Clone tpruvot repo from $TPccminer_repo"
+      git clone $TPccminer_repo ${NVOC_MINERS}/TPccminer/src/
+    fi
+    if ! grep -q $TPccminer_src_hash_ver_8 ${NVOC_MINERS}/TPccminer/src/.git/ORIG_HEAD
+    then
+      echo "Set tpruvot ccminer repo head to $TPccminer_src_hash_ver_8"
+      cd ${NVOC_MINERS}/TPccminer/src/
+      git reset --hard $TPccminer_src_hash_ver_8
+      git fetch origin $TPccminer_src_hash_ver_8
```

Comment:
If you switch to shallow clones this will fail to fetch older commits of the same branch or commits from another branch so most cuda8 miners (old commit) and potential new miners (released in different branch) will fail to checkout.